snapshot/snapshotfs/upload_os_snapshot_windows.go

```diff
@@ -0,0 +1,76 @@
+package snapshotfs
+
+import (
+	"context"
+	"path/filepath"
+
+	"github.com/mxk/go-vss"
+	"github.com/pkg/errors"
+
+	"github.com/kopia/kopia/fs"
+	"github.com/kopia/kopia/fs/localfs"
+	"github.com/kopia/kopia/snapshot/policy"
+)
+
+func osSnapshotMode(p *policy.OSSnapshotPolicy) policy.OSSnapshotMode {
+	return p.VolumeShadowCopy.Enable.OrDefault(policy.OSSnapshotNever)
+}
+
+//nolint:wrapcheck
+func createOSSnapshot(ctx context.Context, root fs.Directory, _ *policy.OSSnapshotPolicy) (newRoot fs.Directory, cleanup func(), err error) {
```

Comment:
nit: can we rename 'err' to something unique here to avoid confusion something like `finalErr`, this would be more obvious in `defer`